addons/website/models/website_menu.py

```diff
@@ -80,8 +81,16 @@ def unlink(self):
     @api.one
     def _compute_visible(self):
         visible = True
+
         if self.page_id and not self.page_id.sudo().is_visible and not self.user_has_groups('base.group_user'):
             visible = False
+
+        if self.group_ids:
+            user = self.env.user
+            # Loops over all groups from the current user. If any of the security groups on the menu.item record
+            # are set on the current user 'visible' will be True, otherwise it will be False.
+            # As the user matches for atleast one security group we assume he/she should have access.
+            visible = all(user.has_group(list(g.get_external_id().values())[0]) for g in self.group_ids)
```

Comment:
`visible = self.group_ids and all(user.has_group(list(g.get_external_id().values())[0]) for g in self.group_ids)`

because `all([]) --> True`

Reply:
This one is done on purpose. This makes the default and no values defaults to True, making everything behave and work out of the box just like it does now. Only explicitly adding items changes it.

Reply:
I have not seen `if self.group_ids`. 
It is not obvious that `all([]) = True` (for math guy it is : because 1^0=1) :-)